### PR TITLE
feat: [WPF] Adds support for ConnectionProfile

### PR DESCRIPTION
**Pull request by ahoefling**
GitHub Issue (If applicable): #6435

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Retrieving the `NetworkConnectivityLevel` on any Skia target always returns `NetworkConnectivityLevel.None` as it is not implemented.

## What is the new behavior?
Now in the WPF Skia target when you request the `NetworkConnectivityLevel` it will return either
- `NetworkConnectivityLevel.InternetAccess`
- `NetworkConnectivityLevel.None`

### Screenshots
**No Internet**
image

**Internet**
image

### Example usage
```c#
var currentConnectivity = NetworkConnectivityLevel.None;
var profile = NetworkInformation.GetInternetConnectionProfile();
if (profile != null)
  currentConnectivity = profile.GetNetworkConnectivityLevel();
```

In this example we can now retrieve the correct connectivity level and `currentConnectivity` will either be `None` or `InternetAccess`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
I want to address the unchecked boxes on the PR checklist
- Docs - I don't think documentation updates are necessary as this applies to the UWP API and doesn't introduce any changes to it
- Unit Tests/UI Tests - This code should be covered by the existing unit tests as I used the Uno Samples app. If there is more work to do on this, I would like some guidance on what I need to do. More than willing to add additional tests.
- Validated PR Screenshots - I am not 100% sure what this means, I have added 2 screenshots demonstrating the functionality with the WPF Skia Uno Sample

Internal Issue (If applicable):
N/A

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Uno.Extensions.Networking.Connectivity
+{
+	internal class ConnectionProfileExtension : IConnectionProfileExtension
+	{
+		private readonly object _connectionProfile;
+		public ConnectionProfileExtension(object owner)
+		{
+			if (!(owner is ConnectionProfile connectionProfile))
+				throw new InvalidOperationException($"Owner of {nameof(ConnectionProfileExtension)} must be a {nameof(ConnectionProfile)}.");
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs:13]:
```suggestion
			if (!(owner is ConnectionProfile connectionProfile))
			{
				throw new InvalidOperationException($"Owner of {nameof(ConnectionProfileExtension)} must be a {nameof(ConnectionProfile)}.");
			}
```

**Review by jeromelaban — APPROVED** [review]
Thanks a lot for your contribution!

File: src/Uno.UWP/Networking/Connectivity/ConnectionProfile.skia.cs [hunk 2]
```diff
@@ -1,17 +1,18 @@
-﻿namespace Windows.Networking.Connectivity
+﻿using Uno.Foundation.Extensibility;
```
[the remark this answers is not in the document]
Comment (ahoefling) [reply src/Uno.UWP/Networking/Connectivity/ConnectionProfile.skia.cs:1]: No problem, I can add this in. For future PRs is this a standard to manually add for all code files or all files that we can add support for?

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Windows.Networking.Connectivity;
+
+namespace Uno.Extensions.Networking.Connectivity
+{
+	internal class ConnectionProfileExtension : IConnectionProfileExtension
```
[the remark this answers is not in the document]
Comment (ahoefling) [reply src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs:7]: Good idea, makes sense to prefix this with `Windows*`

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs (hunk 1 above)
Comment (ahoefling) [reply src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs:13]: Easy to fix, I'll update this on my next code push

I wasn't exactly sure about the coding styles and I couldn't find a document on the contributing guide in GitHub or the documentation in Uno Docs. Is there a style guide somewhere I can reference?

**Comment by ahoefling** [comment]
@jeromelaban thanks for the review feedback, I just pushed new changes that should cover all of your concerns. Let me know if there is anything else I need to do for this PR.

**Comment by jeromelaban** [comment]
All good!

**Review by carldebilly — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@ahoefling Seems the PR failed due to diffignore, can you please rebase on/merge latest master, so that the most up-to-date version is there?

**Comment by jeromelaban** [comment]
No need to rebase, it's just a rebuild that will do.

File: src/Uno.UWP/Networking/Connectivity/ConnectionProfile.skia.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UWP/Networking/Connectivity/ConnectionProfile.skia.cs]: For now we manually add it, until we start the migration to nullable completely.

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Wpf/Extensions/Networking/Connectivity/ConnectionProfileExtension.cs]: There it is: https://github.com/unoplatform/uno/blob/master/doc/articles/contributing/guidelines/code-style.md

**Pull request merged**